File: jApp/src/main/java/j/s/yarlykov/ui/fragmentbased/history/HistoryRVAdapter.java
package j.s.yarlykov.ui.fragmentbased.history;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import j.s.yarlykov.R;
import j.s.yarlykov.data.domain.History;

public class HistoryRVAdapter extends RecyclerView.Adapter<HistoryRVAdapter.ViewHolder> {

    private List<History> dataSource;

    public HistoryRVAdapter(List<History> dataSource) {
        this.dataSource = dataSource;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.history_item, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
        viewHolder.bind(dataSource.get(position));

    }

    @Override
    public int getItemCount() {
        return dataSource.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvDate;
        TextView tvValue;
        ImageView imgLogo;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDate = itemView.findViewById(R.id.tvDate);
            tvValue = itemView.findViewById(R.id.tvValue);
            imgLogo = itemView.findViewById(R.id.ivIcon);
        }

        public void bind(History history) {
            tvDate.setText(history.getDate());
            tvValue.setText(history.getTemperature());
            imgLogo.setImageResource(history.getImg());
        }
    }
}

File: kApp/src/main/java/k/s/yarlykov/ui/fragmentbased/history/HistoryRVAdapter.kt
/**
 * Materials:
 *
 * viewType in onCreateViewHolder^
 * https://forums.bignerdranch.com/t/in-oncreateviewholder-you-pass-in-an-int-called-viewtype/7910
 */

package k.s.yarlykov.ui.fragmentbased.history

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import k.s.yarlykov.R
import k.s.yarlykov.data.domain.History
import k.s.yarlykov.util.Utils.logI

class HistoryRVAdapter(val source: List<History>): RecyclerView.Adapter<HistoryRVAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        logI(this, "onCreateViewHolder")
        return ViewHolder(
                LayoutInflater.from(parent.context).inflate(
                        R.layout.history_item,
                        parent,
                        false
                )
        )

    }

    override fun getItemCount() = source.size

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        logI(this, "onBindViewHolder")
        viewHolder.bind(source[position])
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvDate = itemView.findViewById<TextView>(R.id.tvDate)
        val imgLogo = itemView.findViewById<ImageView>(R.id.ivIcon)
        val tvValue = itemView.findViewById<TextView>(R.id.tvValue)

        fun bind(history: History) = with(history){
            tvDate.text = history.date
            tvValue.text = history.temperature
            imgLogo.setImageResource(history.img)
        }
    }
}